src/bin/pg_autoctl/pgctl.c

```diff
@@ -252,6 +252,66 @@ config_find_pg_ctl(PostgresSetup *pgSetup)
 }
 
 
+/*
+ * ensure_extension_installed ensures that the extension is present in the
+ * given Postgres installation. This does the equivalent of:
+ * ls -l $(pg_config --sharedir)/extension/pg_stat_statements.control
+ */
+bool
+ensure_extension_installed(PostgresSetup *pgSetup, char *extName)
+{
+	char pg_config_path[MAXPGPATH] = { 0 };
+	char extension_path[MAXPGPATH] = { 0 };
+	char share_dir[MAXPGPATH] = { 0 };
+	char extension_control_file_name[MAXPGPATH] = { 0 };
+	char *args[3];
+	Program prog;
+
+	log_info("Checking if the %s extension is installed", extName);
+
+	config_find_pg_ctl(pgSetup);
+	path_in_same_directory(pgSetup->pg_ctl, "pg_config", pg_config_path);
+
+	args[0] = pg_config_path;
+	args[1] = "--sharedir";
+	args[2] = NULL;
+	prog = initialize_program(args, false);
+
+	(void) execute_subprogram(&prog);
+
+	if (prog.returnCode == 0)
+	{
+		if (!prog.stdOut)
+		{
+			log_error("Got empty output from pg_config --sharedir");
+			free_program(&prog);
+			return false;
+		}
+		sscanf(prog.stdOut, "%s", share_dir);
+
+		join_path_components(extension_path, share_dir, "extension");
+		sprintf(extension_control_file_name, "%s.control", extName);
+		join_path_components(extension_path, extension_path, extension_control_file_name);
+		if (!file_exists(extension_path))
+		{
+			log_error("Extension %s is not installed", extName);
+			free_program(&prog);
+			return false;
+		}
+	}
+	else
+	{
+		(void) log_program_output(prog, LOG_INFO, LOG_ERROR);
+		log_error("Failed to run \"%s\" on \"%s\", see above for details",
+				  pg_config_path, pgSetup->pgdata);
```

Comment:
We're not using PGDATA at all in this function, so let's not confuse the users by using it in the error message now...